specification/eventgrid/data-plane/cadl/IotHub.cadl

```diff
@@ -0,0 +1,135 @@
+  statusUpdateTime?: string;
+
+  @doc("An integer that is incremented by one each time the device twin is updated.")
+  version?: integer;
+  //TODO: also here
+
+
+  @doc("The thumbprint is a unique value for the x509 certificate, commonly used to find a particular certificate in a certificate store. The thumbprint is dynamically generated using the SHA1 algorithm, and does not physically exist in the certificate.")
+  x509Thumbprint?: ThumbPrintValue;
+}
+
+@doc("Schema of the Data property of an EventGridEvent for a device connection state event (DeviceConnected, DeviceDisconnected).")
+model DeviceConnectionStateEventProperties {
+  @doc("The unique identifier of the device. This case-sensitive string can be up to 128 characters long, and supports ASCII 7-bit alphanumeric characters plus the following special characters: - : . + % _ &#35; * ? ! ( ) , = @ ; $ '.")
+  deviceId?: string;
+
+  @doc("The unique identifier of the module. This case-sensitive string can be up to 128 characters long, and supports ASCII 7-bit alphanumeric characters plus the following special characters: - : . + % _ &#35; * ? ! ( ) , = @ ; $ '.")
+  moduleId?: string;
+
+  @doc("Name of the IoT Hub where the device was created or deleted.")
+  hubName?: string;
+
+  @doc("Information about the device connection state event.")
+  deviceConnectionStateEventInfo?: DeviceConnectionStateEventInfo;
+}
+
+@doc("Schema of the Data property of an EventGridEvent for a device telemetry event (DeviceTelemetry).")
+model DeviceTelemetryEventProperties {
+  @doc("The content of the message from the device.")
+  body?: object;
+
+  @doc("Application properties are user-defined strings that can be added to the message. These fields are optional.")
+  properties?: Record<string>;
+
+  @doc("System properties help identify contents and source of the messages.")
+  systemProperties?: Record<string>;
+}
+
+@doc("A portion of the properties that can be written only by the application back-end, and read by the device.")
+model DeviceTwinProperties {
+  @doc("Metadata information for the properties JSON document.")
+  metadata?: DeviceTwinMetadata;
+
+  @doc("Version of device twin properties.")
+  version?: float64;
+}
+
+@doc("Metadata information for the properties JSON document.")
+model DeviceTwinMetadata {
+  @doc("The ISO8601 timestamp of the last time the properties were updated.")
+  lastUpdated?: string;
+}
+
+@doc("Information about the device connection state event.")
+model DeviceConnectionStateEventInfo {
+  @doc("Sequence number is string representation of a hexadecimal number. string compare can be used to identify the larger number because both in ASCII and HEX numbers come after alphabets. If you are converting the string to hex, then the number is a 256 bit number.")
+  sequenceNumber?: string;
+}
+
+@doc("Event data for Microsoft.Devices.DeviceCreated event.")
+model IotHubDeviceCreatedEventData extends DeviceLifeCycleEventProperties {}
```

Comment:
Unless this is done only to reduce changes to generated openapi2 documents, this should be a spread. We want to avoid `extends` in Azure unless it is a discriminated type relationship...